arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/Hub.java

```diff
@@ -560,30 +560,27 @@ public void traceContextEnter(MessageFrame frame) {
 
       checkArgument(
           recipientAddress.equals(effectiveToAddress(currentTransaction.getBesuTransaction())));
-      checkArgument(senderAddress.equals(txStack.current().getBesuTransaction().getSender()));
-      checkArgument(isDeployment == transients.tx().getBesuTransaction().getTo().isEmpty());
+      checkArgument(senderAddress.equals(currentTransaction.getBesuTransaction().getSender()));
+      checkArgument(isDeployment == currentTransaction.getBesuTransaction().getTo().isEmpty());
       checkArgument(
-          value.equals(Wei.of(transients.tx().getBesuTransaction().getValue().getAsBigInteger())));
-      checkArgument(frame.getRemainingGas() == transients.tx().getInitiallyAvailableGas());
+          value.equals(
+              Wei.of(currentTransaction.getBesuTransaction().getValue().getAsBigInteger())));
+      checkArgument(frame.getRemainingGas() == currentTransaction.getInitiallyAvailableGas());
 
-      final boolean copyTransactionCallData =
-          !isDeployment && currentTransaction.requiresEvmExecution();
+      final boolean copyTransactionCallData = currentTransaction.copyTransactionCallData();
       if (copyTransactionCallData) {
         callStack.newTransactionCallDataContext(
-            callDataContextNumber(copyTransactionCallData),
-            transients.tx().getBesuTransaction().getData().orElse(Bytes.EMPTY));
+            callDataContextNumber(true), currentTransaction.getBesuTransaction().getData().get());
       }
 
       callStack.newRootContext(
           newChildContextNumber(),
           senderAddress,
           recipientAddress,
           new Bytecode(
-              recipientAddress == null
-                  ? transients.tx().getBesuTransaction().getData().orElse(Bytes.EMPTY)
-                  : Optional.ofNullable(frame.getWorldUpdater().get(recipientAddress))
-                      .map(AccountState::getCode)
-                      .orElse(Bytes.EMPTY)),
+              Optional.ofNullable(frame.getWorldUpdater().get(recipientAddress))
+                  .map(AccountState::getCode)
+                  .orElse(Bytes.EMPTY)),
```

Comment:
Does this work for deployments ? I'm not convinced this will work for deployment transactions.

Reply:
we never entered in the other case, as recipientAddress is set as the effectiveTo address of the tx, so imho, yes.
(Actually, it was a simplification proposed by Intellij, I just verified it was ok)

Reply:
How does the transaction's init code enter the picture i.e. `transients.tx().getBesuTransaction().getData()` ?

Reply:
I think that the issue is with the initial condition. It should be replaced with `tx.to.isEmpty ? ... : ...`

Reply:
Could this explain our ongoing issue with deployment transactions in the ToyWorld v1 ?

Reply:
Hm, no, Besu itself didn't care to execute an instructions. So that's not the explanation.

Reply:
added the initCode in the callFrame in case of deployment